### PR TITLE
Assignment four first draft

### PR DESCRIPTION
**Pull request by kkorn2019**
Will make changes as needed based on comments on other 3 open merge requests.

### CODE REVIEW

File: KaylaKornelis-app/src/main/java/edu/KaylaKornelis/advancedjava/Assignment2/BasicStockService.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import java.math.BigDecimal;
+import java.util.Date;
+
+/**
+ *  This class implements the StockService interface for getting stock data.
+ */
+
+public class BasicStockService implements StockService {
+    
+    private String stockSymbol;
```
Comment (smarks) [inline KaylaKornelis-app/src/main/java/edu/KaylaKornelis/advancedjava/Assignment2/BasicStockService.java]: why are these here?

File: KaylaKornelis-app/src/main/java/edu/KaylaKornelis/advancedjava/Assignment2/StockServiceFactory.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+/**
+ *  This class serves as a Factory for the StockService interface and classes that implement the StockService interface for getting stock data.
+ * Allows creation of new objects without exposing the creation logic to the client.
+ */
+
+public class StockServiceFactory {
+    
+    public StockService getStockService(){
+        /**
```
Comment (smarks) [inline KaylaKornelis-app/src/main/java/edu/KaylaKornelis/advancedjava/Assignment2/StockServiceFactory.java]: needs to be static

**Comment by smarks** [comment]
I am confused a bit here. there seem to be  classes with the same names. I think you are including past week's code rather than merging. we can discuss if this doesn't make sense.

File: src/main/java/edu/KaylaKornelis/advancedjava/Assignment4/BasicStockService.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import org.apache.http.annotation.Immutable;
+
+/**
+ *  This class implements the StockService interface for getting stock data.
+ */
+@Immutable
+public class BasicStockService implements StockService {
+  
+    private Date dateRecorded;
+    private BigDecimal stockPrice;
```
Comment (smarks) [inline src/main/java/edu/KaylaKornelis/advancedjava/Assignment4/BasicStockService.java]: if you initialized you these variables it would work otherwise you are creating a StockQuote with Null values which you don't want to do.

File: src/main/java/edu/KaylaKornelis/advancedjava/Assignment4/StockQuoteApplication.java [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+     * @return a list of StockQuote instances
+     * One for each day in the range specified
+     */
+    @NotNull
+    public static List<StockQuote> createListOfStockQuotes(String symbol, String from, String until){
+        /**
+         * Create a new instance of SimpleDateFormat that will be used to 
+         * parse the string arguments to obtain desired start and end dates
+         */
+        SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/YYYY");
```
Comment (smarks) [inline src/main/java/edu/KaylaKornelis/advancedjava/Assignment4/StockQuoteApplication.java]: whole word variable make easier to read.

File: KaylaKornelis-app/src/main/java/edu/KaylaKornelis/advancedjava/Assignment2/StockServiceFactory.java (hunk 2 above)
Comment (kkorn2019) [reply KaylaKornelis-app/src/main/java/edu/KaylaKornelis/advancedjava/Assignment2/StockServiceFactory.java]: fixed

File: KaylaKornelis-app/src/main/java/edu/KaylaKornelis/advancedjava/Assignment2/BasicStockService.java (hunk 1 above)
Comment (kkorn2019) [reply KaylaKornelis-app/src/main/java/edu/KaylaKornelis/advancedjava/Assignment2/BasicStockService.java]: removed duplicate files

File: src/main/java/edu/KaylaKornelis/advancedjava/Assignment4/StockQuoteApplication.java (hunk 4 above)
Comment (kkorn2019) [reply src/main/java/edu/KaylaKornelis/advancedjava/Assignment4/StockQuoteApplication.java]: fixed

File: src/main/java/edu/KaylaKornelis/advancedjava/AdvancedJavaAssignment4/model/StockQuote.java [hunk 5]
```diff
@@ -1,17 +1,21 @@
...
- * This class in immutable. It holds stock info for a given point time.
+ * This class is immutable. It holds stock info for a given point time.
  */
+@Immutable
 public class StockQuote {
 
-    private Date dateRecorded;
-    private BigDecimal stockPrice;
-    private String stockSymbol;
+    private final Date dateRecorded;
```
Comment (smarks) [inline src/main/java/edu/KaylaKornelis/advancedjava/AdvancedJavaAssignment4/model/StockQuote.java:16]: nice

File: src/main/java/edu/KaylaKornelis/advancedjava/Assignment4/BasicStockService.java (hunk 3 above)
Comment (kkorn2019) [reply src/main/java/edu/KaylaKornelis/advancedjava/Assignment4/BasicStockService.java]: initialized these variables

File: src/main/java/edu/KaylaKornelis/advancedjava/AdvancedJavaAssignment4/service/BasicStockService.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    @Override
+    public List<StockQuote> getQuote(String symbol, Calendar from, Calendar until, Interval interval){
+        /**
+         * Create new array list to hold list of stock quotes
+         */
+        List<StockQuote> listOfStockQuotes = new ArrayList<>();
+        
+        /**
+         * Create new stockQuote for the given symbol and add it to the array list
+         */
```
Comment (smarks) [inline src/main/java/edu/KaylaKornelis/advancedjava/AdvancedJavaAssignment4/service/BasicStockService.java:85]: good clean up

**Review by smarks — APPROVED** [review]
Good clean up here. Maybe focus on next assignment now.

File: src/main/java/edu/KaylaKornelis/advancedjava/AdvancedJavaAssignment4/model/Interval.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    
+    /**
+     *  Create a new  Interval instance
+     * @param requestedInterval the time frame to obtain quotes for 
+     */
+    @NotNull
+    private Interval(String requestedInterval){
+        this.interval = requestedInterval;
+    }
+
```
Comment (kkorn2019) [reply src/main/java/edu/KaylaKornelis/advancedjava/AdvancedJavaAssignment4/model/Interval.java:39]: fixed

Comment (smarks) [inline src/main/java/edu/KaylaKornelis/advancedjava/AdvancedJavaAssignment4/model/Interval.java:39]: don't forget javadoc
